src/test/run-pass/async-fn-size-moved-locals.rs

```diff
@@ -0,0 +1,97 @@
+// Test that we don't duplicate storage for futures moved around in .await, and
```

Comment:
Please move this file into `src/test/ui/async-await/` (there's another file `async-fn-size.rs` you made before also, please move that too)

Reply:
We also have `src/test/run-pass/async-await/` which I moved some tests into, is there a reason to prefer one over another?

Reply:
There's no `src/test/run-pass/async-await/`? (the `run-pass` folder will be going away soon hopefully...)

Reply:
Oh right, that's in #61818 which hasn't merged yet.

Reply:
Would be good to change #61818 as well :)

Reply:
I'd rather do it in a separate change at this point. All the size-related tests are in `run-pass` today.

Can you link to any discussion around why `run-pass` is going away?

Reply:
I don't have a link on hand, but basically categorizing by pass/fail is suboptimal; ideally we categorize by semantic area, e.g. async-await, traits, etc.

Reply:
Sure, if the goal is to subsume everything with `ui` and then get rid of the `ui` part :)

Reply:
However currently we are still sticking to the run-pass/ui distinction. Also see https://github.com/rust-lang/rust/pull/61701#issuecomment-500305911.

Reply:
https://github.com/rust-lang/rust/pull/61755 is almost merged, so it should be ok to move tests to ui.

Reply:
Awesome. :)